scripts.js

```diff
@@ -1,127 +1 @@
-    buttonOk.setAttribute('data-date', cell.getAttribute('data-date'));
-    var modalTitle = modalFrame.getElementsByClassName('modal-title')[0];
-    modalTitle.innerHTML = buttonOk.getAttribute('data-date');
-}
-
-function deleteFromStorage(){
-    let selectedDay = buttonOk.getAttribute('data-date');
-    let calendarElements = document.querySelector("#calendar");
-    let selectedCell = calendarElements.querySelector(`[data-date = "${selectedDay}"]`);
-    localStorage.removeItem(selectedDay);
-    selectedCell.classList.remove("noted");
-}
-
-function next(){
-    if (currentMonth === 11){
-        currentYear +=1;
-    }
-    currentMonth = (currentMonth + 1) % 12;
-    showCalendar (currentMonth, currentYear);
-}
-
-function previous(){
-    if (currentMonth === 0){
-        currentYear -=1;
-        currentMonth = 11;
-    } else {
-        currentMonth -= 1;
-    }
-    showCalendar (currentMonth, currentYear);
-}
-
-function jump(){
-    currentMonth = parseInt(selectMonth.value);
-    currentYear = parseInt(selectYear.value);
-    showCalendar (currentMonth, currentYear);
-}
-
-function showCalendar(month, year){
-    let firstDay = (new Date(year, month)).getDay();
-    if (firstDay === 0) firstDay = 7;  // для начала недели с Пн
-    let daysInMonth = (new Date(year, month + 1, 0)).getDate();
-    let tbl = document.getElementById("calendar-body");
-    tbl.innerHTML = "";
-    monthAndYear.innerHTML = months[month] + " " + year;
-    selectYear.value = year;
-    selectMonth.value = month;
-
-    //создание ячеек
-    let date = 1;
-    for (let i = 0; i < 6; i++) {
-        let row = document.createElement("tr");
-
-        for (let j = 1; j <= 7; j++){
-            if (i === 0 && j < firstDay){ //создание пустых ячеек до нужного дня недели
-                let cell = document.createElement("td");
-                let cellText = document.createTextNode("");
-                cell.appendChild(cellText);
-                row.appendChild(cell);
-            } else if (date > daysInMonth) { //проверка конца месяца
-                break;
-            } else { //заполнение датами с нужного дня недели
-                let cell = document.createElement("td");
-                let cellText = document.createTextNode(date);
-                cell.classList.add("calendar-days"); 
-                if (date === today.getDate() && year === today.getFullYear() && month === today.getMonth()){
-                    cell.classList.add("highlighted"); //подсветить сегодня
-                }
-                
-                let atribute = year + '-' + (month+1) + '-' + date;
-                cell.dataset.date = atribute;//добавление дата атрибута для каждой даты
-                if (localStorage.getItem(atribute) !== null){
-                    cell.classList.add("noted");  //подсвечивать дни с заметками
-                }
-                cell.appendChild(cellText);
-                row.appendChild(cell);
-                date++;

```

Comment:
1) I would create separate functions for the creation of days data and DOM elements
2) create separate functions for every entity to increase readability of code
```
const createMonth = () => {
   // ...some code here
   
   // you should invent some format of your data here, for example, an array like this
   return [
     [
         { date: '01.01.2020', isToday: true, hasNotes: false }, // day 1
        { date: '01.01.2020', isToday: true, hasNotes: false }, // day 1
        ... // etc
      ], // week 1
     [...] // week 2
   ]; 
} 

const createDayCell = (dayData) => {
}

const createWeekRow = (weekData) => {
}

const createMonthTable = (monthData) => {
}
```

Reply:
also, it would be nice if you will work with arrays and `map`/`forEach` functions instead of `for` loops while you will refactor this